onnx/defs/schema.cc

```diff
@@ -22,14 +22,18 @@ namespace ONNX_NAMESPACE {
 // -1 means ONNX schema hasn't been loaded yet
 // 0 means all versions of ONNX schema have been loaded
 // Other positive integer means the ONNX schemas for the specified version have been loaded
+#ifdef __ONNX_DISABLE_STATIC_REGISTRATION
 int OpSchemaRegistry::loaded_schema_version = -1;
+#else
+int OpSchemaRegistry::loaded_schema_version = 0;
```

Comment:
> Found another issue, loaded_schema_version [here](https://github.com/onnx/onnx/blob/main/onnx/defs/schema.cc#L25) should probably be assigned based on __ONNX_DISABLE_STATIC_REGISTRATION, no?

https://github.com/onnx/onnx/blob/9184c2989b2fb3e555680985c3c3a5aa19555454/onnx/defs/operator_sets.h#L1137 loaded_schema_version will be set after `RegisterOnnxOperatorSetSchema` and initially it will be -1, which means onnx hasn't registered any opset yet. Not sure why setting 0 here?

Reply:
This ties to the intention [here](https://github.com/onnx/onnx/pull/5221#pullrequestreview-1442771211), where a runtime might, at some point, use `GetLoadedSchemaVersion` to understand what ONNX schemas are loaded. In the case where all schema is registered at start (by default), the call would otherwise returns `-1` to runtime, which is a wrong information suggesting nothing is loaded.

Reply:
I didn't change this line in [641cd5a](https://github.com/onnx/onnx/pull/5221/commits/641cd5a5e8b43fd4a29f741a00a1790166ecc6ca) as explained above. Let me know what you think. If I missed something or it doesn't make sense, I'll make change.

Reply:
Not quite follow here -- let me organize the flows for these two options:
- Turn on ONNX_DISABLE_STATIC_REGISTRATION: runtime might register onnx schema for certain opset_version multiple times
  - Initially loaded_schema_version=-1; which means no schema has been registered yet
  - runtime will further call `RegisterOnnxOperatorSetSchema` with certain opset_version to register specified opset_version (and `loaded_schema_version` will be set as that specified version).
  - (Optional) runtime may call `DeregisterOnnxOperatorSetSchema` to clear all registration (and `loaded_schema_version` will be set as -1) and then call `RegisterOnnxOperatorSetSchema` with another opset_version  (and `loaded_schema_version` will be changed accordingly).
- Turn off ONNX_DISABLE_STATIC_REGISTRATION: runtime will just register onnx schema once
  - Initially loaded_schema_version=-1; which means no schema has been registered yet
  - runtime will further call `RegisterOnnxOperatorSetSchema` to register all onnx schema and then `loaded_schema_version` will be set as 0.

Thus, I think anyway OpSchemaRegistry::loaded_schema_version will initially declared as -1 by default, which lets onnxruntime know there is no schema has been registered yet. Did I missing something obvious?

Reply:
Understood - agree. Though in the "turned off" case, runtime won't normally explicitly call `RegisterOnnxOperatorSetSchema`, right? Updated PR to bring it back to the correct `-1` value.